java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -405,4 +418,13 @@ public ObjectConstraint getConstraintWithStatus(SymbolicValue value, Object aSta
     }
     return null;
   }
+
+  MethodYield yield() {
+    return yield;
+  }
+
+  ProgramState initYield(int size) {
+    Preconditions.checkState(this.yield == EMPTY_YIELD);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)